jsEngine/settings/Settings.ts

```diff
@@ -19,19 +27,73 @@ export class JsEnginePluginSettingTab extends PluginSettingTab {
 		this.plugin = plugin;
 	}
 
+	private async listJSfilesInDirectory(directory: string): Promise<TFile[]> {
+		const { vault } = this.app;
+		return (await vault.adapter.list(directory)).files
```

Comment:
I would prefer to get the `TFolder` corresponding to the folder path and then collect the JS files via `TFolder#children`